modules/launcher/src/main/java/org/mule/module/launcher/artifact/AbstractArtifactClassLoader.java

```diff
@@ -60,9 +69,62 @@ public void dispose()
                 logger.error(e);
             }
         }
+
+        LeakCleaner leakCleaner = getLeakCleanerInstance();
+        if (leakCleaner != null)
+        {
+            leakCleaner.clean();
+        }
+
         super.dispose();
     }
 
+    public String getLeakCleanerClassName()
+    {
+        return StringUtils.defaultString(leakCleanerClassName, DEFAULT_LEAK_CLEANER_CLASS);
+    }
+
+    public void setLeakCleanerClassName(String leakCleanerClass)
+    {
+        this.leakCleanerClassName = leakCleanerClass;
+        leakCleaner = null;
+    }
+
+    public LeakCleaner getLeakCleanerInstance()
+    {
+        if(leakCleaner==null)
+        {
+            leakCleaner = createLeakCleanerInstance(getLeakCleanerClassName());
+        }
+        return leakCleaner;
+    }
+
+    private LeakCleaner createLeakCleanerInstance(String className)
```

Comment:
The method can be protected, but I can not inject the LeakCleaner, that instance must be created in this class loader in this strange way. If I inject it from outside the whole trick won't work.